src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach[||](var a in array) /* comment */;
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++) /* comment */;
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment4()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach[||](var a in array) Console.WriteLine(a); // test
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++)
+        {
+            var a = array[i];
+            Console.WriteLine(a); // test
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment5()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array) /* test */ Console.WriteLine(a); 
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++) /* test */
+        {
+            var a = array[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment6()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array) 
+            /* test */ Console.WriteLine(a); 
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++)
+        {
+            var a = array[i];
+            /* test */
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment7()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        // test
+        foreach[||](var a in new int[] { 1, 3, 4 })
+        {
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        // test
+        var {|Rename:list|} = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < list.Length; i++)
```

Comment:
why are there two rename annotations here?  How does that actually work in practice?  It seems bad that we're placing two annotations.

Reply:
I wasn't sure whether annotation two works or not. I tried and it worked fine. so I kept rather than tracking whether I already added annotation or not. simpler. the system will chose the first one in the tree.
...
it seems very restrictive though that rename session can only support 1 though. since all generated identifiers we probably want to offer rename sessions.

Reply:
Fair enough.  If you file this, can you open an issue to make this work.  i.e. "Inline rename should support multiple rename sessions for refactorings that introduce multiple variables".